[ZT] Add new Gateway log fields (#7658)
cloudflare/cloudflare-docs: content/cloudflare-one/analytics/logs/gateway-logs/_index.md

Enterprise users can generate more detailed logs with [Logpush](/cloudflare-one/analytics/logs/logpush/).

## Selective logging

By default, Gateway logs all events, including DNS queries and HTTP requests that are allowed and not a risk. You can choose to disable logs or only log blocked requests. To customize what type of events are recorded, log in to your [Zero Trust dashboard](https://one.dash.cloudflare.com/) and go to **Settings** > **Network**. Under **Activity Logging**, indicate your DNS, Network, and HTTP log preferences.

These settings will only apply to logs displayed on the Zero Trust dashboard. Logpush data is unaffected.

## DNS logs

### Explanation of the fields

{{<table-wrap>}}

| **Field**             | **Description**                                                                                                                        |
| --------------------- | -------------------------------------------------------------------------------------------------------------------------------------- |

> | **DNS**               | Name of the domain that was queried.                                                                                                   |

Reviewer: Nit: Are these fields listed in any particular order? (for example, in order of how they appear in the UI)

| **Port**              | Port that was used to make the DNS query.                                                                                              |
| **Source IP Country** | Country code of the DNS query.                                                                                                         |
| **Resolver IP**       | Public IP address of the DNS resolver.                                                                                                 |
| **Protocol**          | Protocol that was used to make the DNS query (for example, `https`).                                                                   |
| **Query Type**        | The type of DNS query. [This page](https://en.wikipedia.org/wiki/List_of_DNS_record_types) contains a list of all the DNS query types. |
| **Resolver Decision** | The reason why Gateway applied a particular **Action** to the request. Refer to the [list of resolver decisions](#resolver-decisions). |
| **DNS Location**      | User-configured location from where the DNS query was made.                                                                            |

> | **Policy Name**       | Policy name of matched policy (if there is one).                                                                                       |

Reviewer: ```suggestion
| **Policy Name**       | Name of matched policy (if there is one).                                                                                       |
```

| **Policy ID**         | ID of the matched policy (if there is one).                                                                                            |

> | **Categories**        | The content categories that the domain belongs to.                                                                                     |

Reviewer: Nit: I noticed that some of these descriptions start with "The" and others do not.

| **Email**             | The email address of the user who made the DNS query. This is generated by the WARP client.                                            |
| **User ID**           | The ID of the user who made the DNS query. This is generated by the WARP client.                                                       |
| **Device ID**         | The ID of the device that made the DNS query. This is generated by the WARP client.                                                    |

> | **Location ID**       | Location ID number matched                                                                                                             |

Reviewer: ```suggestion
| **Location ID**       | The ID of the [DNS location](/cloudflare-one/connections/connect-devices/agentless/dns/locations/) where the query originated.                                                                                             |
```

> | **Device Name**       | The name of the device of the user who made the DNS request.                                                                           |

Reviewer: ```suggestion
| **Device Name**       | The name of the device that made the DNS request.                                                                           |
```

| blockedAlwaysCategory  | Domain or hostname is always blocked by Cloudflare.         |
| allowedOnNoLocation    | Allowed because query did not match a Gateway DNS location. |
| allowedOnNoPolicyMatch | Allowed because query did not match a policy.               |
| overrideForSafeSearch  | Response was overridden by a SafeSearch policy.             |
| overrideApplied        | Response was overridden by an Override policy.              |

{{</table-wrap>}}

## Network logs

### Explanation of the fields

{{<table-wrap>}}

| Field                      | Description                                                                                                                       |
| -------------------------- | --------------------------------------------------------------------------------------------------------------------------------- |
| **Source IP**              | The IP address of the user sending the packet.                                                                                    |
| **Destination IP**         | The IP address of the packet’s target.                                                                                            |

> | **Action**                 | The Gateway action taken based on the first rule that matched. For example: Allowed, Blocked, Bypass, etc.                        |

Reviewer: ```suggestion
| **Action**                 | The Gateway [Action](/cloudflare-one/policies/filtering/dns-policies/#actions)  taken based on the first rule that matched  (for example, `Allow` or `Block`).            |
```

| **Source IP Country**      | Country code for the packet source.                                                                                               |
| **Destination Port**       | The destination port number for the packet.                                                                                       |
| **Destination IP Country** | Country code for the packet destination.                                                                                          |
| **Protocol**               | The protocol over which the packet was sent.                                                                                      |
| **SNI**                    | The host whose Server Name Indication (SNI) header Gateway will filter traffic against.                                           |
| **Virtual Network**        | [Virtual network](/cloudflare-one/connections/connect-apps/private-net/tunnel-virtual-networks/) that the client is connected to. |
| **Categories**             | Category or categories associated with the packet.                                                                                |

> | **Policy Name**            | Policy name of matched policy (if there is one).                                                                                  |

Reviewer: ```suggestion
| **Policy Name**            | Name of matched policy (if there is one).                                                                                  |
```

| **Policy ID**              | The ID of the policy enforcing the decision Gateway made.                                                                         |
| **Email**                  | The email address of the user sending the packet. This is generated by the WARP client.                                           |
| **User ID**                | The ID of the user sending the packet. This is generated by the WARP client.                                                      |
| **Device ID**              | The ID of the device that sent the packet. This is generated by the WARP client.                                                  |

> | **Device Name**            | The name of the device of the user who made the DNS request                                                                       |

Reviewer: ```suggestion
| **Device Name**            | The name of the device that made the DNS query.                                                                  |
```

| ---------------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
| **Host**                     | The hostname in the HTTP header for the HTTP request.                                                                                                                                                                                                                                                                                       |
| **Email**                    | Email address of the user who made the DNS query. This is generated by the WARP client.                                                                                                                                                                                                                                                     |

> | **Action**                   | The Gateway action taken based on the first rule that matched. For example: Allowed, Blocked, Bypass, etc.                                                                                                                                                                                                                                  |

Reviewer: ```suggestion
| **Action**                   | The Gateway [Action](/cloudflare-one/policies/filtering/dns-policies/#actions)  taken based on the first rule that matched  (for example, `Allow` or `Block`).                                                                                                                                                                                                                                             |
```

| **Destination Port**         | Port of the destination requested.                                                                                                                                                                                                                                                                                                          |
| **Destination IP Country**   | Country code of the destination requested.                                                                                                                                                                                                                                                                                                  |
| **Blocked file reason**      | Reason why the file was blocked if a file transfer occurred or was attempted.                                                                                                                                                                                                                                                               |

> | **Policy Name**              | Policy name of the matched policy (if there is one).                                                                                                                                                                                                                                                                                        |

Reviewer: ```suggestion
| **Policy Name**              | Name of the matched policy (if there is one).                                                                                                                                                                                                                                                                                      |
```

| **Policy Description**       | Description of the matched policy (if there is one).                                                                                                                                                                                                                                                                                        |
| **Email**                    | The email address of the user who made the DNS query. This is generated by the WARP client.                                                                                                                                                                                                                                                 |
| **User ID**                  | The ID of the user who made the DNS query. This is generated by the WARP client.                                                                                                                                                                                                                                                            |

> | **Device Name**              | The name of the device of the user who made the DNS request                                                                                                                                                                                                                                                                                 |

Reviewer: ```suggestion
| **Device Name**              | The name of the device that made the DNS query.                                                                                                                                                                                                                                                                               |
```
